### PR TITLE
[storage] Store `enum Transaction` instead of `struct SignedTransaction`

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation
We are gonna use the former everywhere. `SignedTransaction` used to represent all possible types of transactions but will be only the "user created" variant of it.

part of the effort around #1173 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Unit tests.
## Related PRs
#1175  introduced the type

### CODE REVIEW

File: types/src/transaction.rs [hunk 1]
```diff
@@ -547,7 +547,7 @@ impl SignedTransactionWithProof {
     ///   2. And this signed transaction has the same `version`, `sender`, and `sequence_number` as
     /// indicated by the parameter list. If any of these parameter is unknown to the call site that
     /// is supposed to be informed via this struct, get it from the struct itself, such as:
-    /// `signed_txn_with_proof.version`, `signed_txn_with_proof.signed_transaction.sender()`, etc.
+    /// `signed_txn_with_proof.version`, `signed_txn_with_proof.transaction.sender()`, etc.
```
Comment (wqfish) [inline types/src/transaction.rs]: But you are not renaming the field in this commit?

File: storage/libradb/src/schema/transaction/mod.rs [hunk 2]
```diff
@@ -43,14 +36,13 @@ impl KeyCodec<TransactionSchema> for Version {
     }
 }
 
-impl ValueCodec<TransactionSchema> for SignedTransaction {
+impl ValueCodec<TransactionSchema> for Transaction {
     fn encode_value(&self) -> Result<Vec<u8>> {
-        let event: types::proto::types::SignedTransaction = self.clone().into();
```
Comment (wqfish) [inline storage/libradb/src/schema/transaction/mod.rs]: `event` :)

File: types/src/unit_tests/transaction_test.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 use crate::{
     account_address::AccountAddress,
-    transaction::{RawTransaction, Script, Transaction, SignedTransaction, TransactionPayload},
+    transaction::{RawTransaction, Script, SignedTransaction, Transaction, TransactionPayload},
```
Comment (wqfish) [inline types/src/unit_tests/transaction_test.rs]: this should go to another commit?

File: storage/libradb/src/schema/transaction/mod.rs (hunk 2 above)
Comment (msmouse) [reply storage/libradb/src/schema/transaction/mod.rs]: not me.. not me.. 🤡 https://github.com/libra/libra/commit/ab4b9ad20bc458a153f35ecd4175804c1f1d6199

File: types/src/unit_tests/transaction_test.rs (hunk 3 above)
Comment (msmouse) [reply types/src/unit_tests/transaction_test.rs]: yup. let's see some serious fixups.

**Comment by runtian-zhou** [comment]
Thanks! IMHO we should add more tests to make sure that adding more variants to the enum will not break serialization but I guess we can do that later.

**Comment by msmouse** [comment]
the fixups seem to have confused the build. I'm gonna need to force push, sorry.
